source/ui/Components/Output/SevenSegment/SevenSegment.qml

```diff
@@ -0,0 +1,180 @@
+        onMemoryChanged: {
+            var _baseAddress = outputComponent.getOutputItem(outputItemIndex)["baseAddress"];
+            var numberOfDigits = outputComponent.getOutputItem(outputItemIndex)["numberOfDigits"];
+            // Check if the memory address that was changed (at least partly) belongs to
+            // the output item's source space.
+            if ((address+length) >= _baseAddress && (address <= (_baseAddress+(numberOfDigits)))) {
+                updateContent(_baseAddress);
+            }
+        }
+        // Sent when any item's settings where updated.
+        onOutputItemSettingsChanged: {
+            updateContent(outputComponent.getOutputItem(outputItemIndex)["baseAddress"]);
+            sevenSegmentSettingsWindow.updateSettings();
+        }
+    }
+
+    // Update the output item's content (there may already be some initial values in memory).
+    Component.onCompleted: {
+        updateContent(outputComponent.getOutputItem(outputItemIndex)["baseAddress"]);
+    }
+
+    // Called from outside by the output tab view to signal that the settings button for the current
+    // output item was pressed.
+    onSettingsButtonPressed: {
+        sevenSegmentSettingsWindow.show();
+    }
+
+
+    // ListView for seven segment digits.
+    ListView {
+        id: digitsListView
+
+        anchors.centerIn: parent
+        height: parent.height - 10
+        spacing: 10
+        width: (height / 2) * count + (spacing) * (count - 1)
+
+        orientation: ListView.Horizontal
+        interactive: false
+
+        model: ListModel {
+            id: sevenSegmentDigitsModel
+        }
+
+        delegate: sevenSegmentDigit
+    }
+
+    // Delegate for the ListView displaying each digit.
+    Component {
+        id: sevenSegmentDigit
+
+        SevenSegmentDigit {
+            height: parent.height
+            width: height / 2
+            thickness: height / 10
+
+            color: "#00000000"  // Background Color
+            digitActiveColor: digitsActiveColor
+            digitInactiveColor: digitsInactiveColor
+            textColor: "black"
+
+            segment0Activated: segment0
+            segment1Activated: segment1
+            segment2Activated: segment2
+            segment3Activated: segment3
+            segment4Activated: segment4
+            segment5Activated: segment5
+            segment6Activated: segment6
+
+            onSegmentPressed: {
+                var segmentIdentifier = "segment" + segmentIndex;
+                sevenSegmentDigitsModel.setProperty(index, segmentIdentifier, !sevenSegmentDigitsModel.get(index)[segmentIdentifier]);
+                var memoryContent = [];
+                for (var digitIndex = (sevenSegmentDigitsModel.count -1); digitIndex >= 0; --digitIndex) {
+                    memoryContent.push(sevenSegmentDigitsModel.get(digitIndex)["segment0"]);
+                    memoryContent.push(sevenSegmentDigitsModel.get(digitIndex)["segment1"]);
```

Comment:
You could just loop from 0 to 6 and push `memoryContent.push(sevenSegmentDigitsModel.get(digitIndex)["segment" + index]);`